src/PointwiseFunctions/Punctures/AdmIntegrals.cpp

```diff
@@ -0,0 +1,28 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/Punctures/AdmIntegrals.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Punctures {
+
+void adm_mass_integrand(const gsl::not_null<Scalar<DataVector>*> result,
+                        const Scalar<DataVector>& field,
+                        const Scalar<DataVector>& alpha,
+                        const Scalar<DataVector>& beta) {
+  get(*result) =
```

Comment:
I would do:
```cpp
get(*result) =  pow<7>(get(alpha) * (get(field) + 1.) + 1.);
get(*result) = 0.5 / M_PI * get(beta) / get(*result);
```
I'm worried that `pow<7>` will expand into repeated nested expressions (which it should to avoid an allocation)

Reply:
good call. I did this now:

```cpp
  get(*result) = get(alpha) * (get(field) + 1.) + 1.;
  get(*result) = pow<7>(get(*result));
  get(*result) = 0.5 / M_PI * get(beta) / get(*result);
```